grakn-engine/src/main/java/ai/grakn/engine/controller/response/ExplanationBuilder.java

```diff
@@ -59,7 +59,8 @@
                         .flatMap(unifier -> rewrittenQuery.materialise(answer.unify(unifier)))
                         .findFirst().orElse(new QueryAnswer());
             }
-            explanation.add((Answer) printer.graqlString(false, inferredAnswer));
+            // TODO: this looks like a bug: the return type is not `Answer`
```

Comment:
Not a bug just poor design. In the context of the explanation builder the return type is actually always an answer.

Reply:
Thanks! I get it. So `JacksonPrinter` returns `Object` and therefore sometimes it is an `Answer`. In this context, we know that it must be.

@fppt suggested we can make this a bit clearer by making this method take a `Printer<Object>` or `JacksonPrinter` explicitly.